### PR TITLE
initial commit of color_sliders.py

### PR DESCRIPTION
**Pull request by tobyhodges**
I've changed the name of the script to better match your naming conventions. At my end it's been called colorPicker.py so far\*.

I tested the script on my system and it runs just fine. I get a warning:  
    `No handlers could be found for logger "bokeh.resources"`  
but this seems to be harmless.

The only big difference that I can see in terms of code style is my inclusion of a namespace for bokeh.plotting  e.g  
    `from bokeh import plotting as bkplt`  
    `source = bkplt.ColumnDataSource(# etc etc`  
which I use because it seems more explicit to me. Happy to change if you would prefer absolute consistency.

\*TBH I think the new name is more descriptive - color picking was always the intended end goal of this script, but I haven't yet had time to devote to figuring out a way to get the two plots to interact!

### CODE REVIEW

File: examples/plotting/file/color_sliders.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+p2.axis.major_label_text_color = None
+p2.axis.major_tick_line_color = None
+p2.axis.minor_tick_line_color = None
+
+layout = bkplt.hplot(
+	vform(red_slider, green_slider, blue_slider),
+	vform(p1, p2)
+)
+
+bkplt.output_file("colorSliders.html")
```
Comment (damianavila) [inline examples/plotting/file/color_sliders.py]: the name has to match the filename + .html, so our testing engine pick the correct one...
So, this has to be `"color_sliders.html"`

Comment (tobyhodges) [inline examples/plotting/file/color_sliders.py]: Thanks @damianavila. I’ve made that change and pushed to the 2270_rgb_sliders_example branch.


> On 12 May 2015, at 14:18, Damián Avila <notifications@github.com> wrote:
> 
> In examples/plotting/file/color_sliders.py <https://github.com/bokeh/bokeh/pull/2275#discussion_r30127575>:
> 
> > +p1.axis.major_tick_line_color = None
> > +p1.axis.minor_tick_line_color = None
> > +
> > +p2.ygrid.grid_line_color = None
> > +p2.xgrid.grid_line_color = None
> > +p2.axis.axis_line_color  = None
> > +p2.axis.major_label_text_color = None
> > +p2.axis.major_tick_line_color = None
> > +p2.axis.minor_tick_line_color = None
> > +
> > +layout = bkplt.hplot(
> > +	vform(red_slider, green_slider, blue_slider),
> > +	vform(p1, p2)
> > +)
> > +
> > +bkplt.output_file("colorSliders.html")
> the name has to match the filename + .html, so our testing engine pick the correct one...
> So, this has to be "color_sliders.html"
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/bokeh/bokeh/pull/2275/files#r30127575>.
> 


--

Dr Toby Hodges

email: tbyhdgs@gmail.com
twitter: @tbyhdgs
github: github.com/tobyhodges

**Comment by tobyhodges** [comment]
Commented on this by email but it seems to have been lost in the tubes, so I'll reproduce here at risk of duplicating:

Thanks @damianavila. I’ve made that change and pushed to the 2270_rgb_sliders_example branch.

**Comment by jakirkham** [comment]
@tobyhodges, I'm not sure how much Python experience you have had; so, I apologize if I am merely pointing out the obvious, but Python 3 made some breaking changes. One of the most notable is the behavior of `range` has changed to be that of `xrange`. Hence, the complaint that this cannot be serialized to JSON. There are many ways that this can be worked around to ensure Python 2/3 compatibility. The easiest is probably to do the following change `range(...)` to `list(range(...))`. This should work on both versions thus fixing the test failures.

**Comment by damianavila** [comment]
Thanks @jakirkham to guide @tobyhodges into some fixes.... very appreciated!!! 
@tobyhodges, also try to use 4 space indentation (avoid tabs).

Also please make the imports explicitly but in this form:

```python
from bokeh.plotting import figure, show, output_file
```

Thanks!!

**Comment by tobyhodges** [comment]
Thanks @jakirkham and @damianavila. I'm unfamiliar with the TravisCI testing system, otherwise I would've been tracking down the `range()` errors myself! I had guessed that you might want the imports changed as well. Should be quick fixes that I can do today.

**Comment by damianavila** [comment]
LGTM, merging soon...

**Comment by damianavila** [comment]
Can I ask for one more change? Since you are importing from models, can you import `ColumnDataSource` from models?

**Comment by tobyhodges** [comment]
Done. To help my general understanding, can you tell me why you prefer it from `bokeh.models`?

**Comment by damianavila** [comment]
Yes, `ColumnDataSource` "belongs" to model: http://bokeh.pydata.org/en/latest/docs/reference/models.html?highlight=columndatasource#bokeh.models.sources.ColumnDataSource
We make it available in `plotting` too, just to let people write less lines when they are using the plotting API: https://github.com/bokeh/bokeh/blob/master/bokeh/plotting.py#L16
Since you are already importing other things from `bokeh.models`, I think is proper to import `ColumnDataSource` from there too... makes sense?

**Comment by tobyhodges** [comment]
Yeah, makes perfect sense! Thanks @damianavila!

**Comment by damianavila** [comment]
Really nice contribution, thanks @tobyhodges !!!

**Comment by tobyhodges** [comment]
My pleasure! Really good for me to start getting the hang of PRs and contributing to a larger project - thanks for your help & guidance everyone.

**Comment by jakirkham** [comment]
@tobyhodges, again sorry if it was already obvious. Yeah, having a CI is pretty cool. It makes it that much easier to maintain software, especially, with large numbers of contributors.

**Pull request merged**